vision.md

```diff
@@ -0,0 +1,39 @@
+# Vision of the buildfarmer payload
+Subset of tasks defined to increase the quality of the ROS/Gazebo code by actively tracking and categorizing problems shipped with new code in a timely manner, making it easier for the development teams to:
+* Fix most errors when they are introduced.
+* Decide if an error that pops up in CI is a flaky test already existing in the codebase.
+* Identify how often an error occurs and decide how to proceed.
+
+## Internal process and rationale behind it
+
+We already have a priority document explaining the order of relevancy of the possible errors, this is a new set of standards to define what we expect of the internal buildfarmer process:
+
+1. Auto-log any error appearing in the nightlies, even when there is no one around. This could serve future investigations.
+2. Any person can become a Buildfarmer Payload in case of necessity, this implies:
+   * Basic duties should be easy to automate, even scripted.
+   * Every Buildfarmer Payload must log/communicate internally any actions completed in the name of the role, so other folks can make progress on top of previous actions.
+   * There will be a public log with investigations related to the Buildfarmer Payload duties.
+   * Actions related to Buildfarmer Payload will be differentiated from actions related to the infra team, it is not resposability of the role to investigate infra issues, instead those will be escalated.
+3. Core Team and community developers should be able to check if an existent error in his build it’s also happening in the nightlies CI.
+4. Flaky tests will be categorized based on how often they appear
```

Comment:
Same comment I did on the document:

"How often they appear" in which timestamp? 

I think it is important to define in which time frame we're talking about (2 months, 1 month, 2 weeks), or in how many builds do we want to measure the percentage.

E.g., for a 1-month timeframe of a ROS nightly job: 5% is 1.55 builds, 15% is 4.65 builds.

In Gazebo, most of the jobs run each four days. We would need to calculate the flakiness in almost 3 months to maybe get a 5% flaky ratio.

Also, our current workflow is priotirizing the "Number of failed builds" over the "Flaky ratio".

Reply:
Been thinking a bit about this, here's the rationale I think we should follow in terms of the number of builds. I'm thinking in terms of 2 weeks of ROS 2 nightlies, which is around 14 builds.

3 errors in 14 builds =  21%
2 errors in 14 builds = 14%
1 error in 14 build = 7%

probably 20% is a good mark to start pushing actively for issues, and disabling them as needed. Anything below 7% should be ignored in terms of investigations, and anything in the middle should be help wanted to the developer teams.

> In Gazebo, most of the jobs run each four days. We would need to calculate the flakiness in almost 3 months to maybe get a 5% flaky ratio.

Hopefully we can improve that in the future, in the meantime the rationale to categorize the flakiness shouldn't vary. When calculating flakiness, we will be taking into account last month test failures and I added an extra section about prioritizing consecutive failures. 

The discussion about how to categorize the flakiness of an error will not be used to categorize new errors, as we simply won't have enough info to consider them flaky.